### ISSUE TITLE
[docs][Tree View] Add a11y/keyboard navigation documentation

### ISSUE THREAD

**Issue by flaviendelangle** [mui/mui-x#12435]
Add `Tree View` accessibility documentation page.

Inspiration can be taken from [`Tree View` page](https://mui.com/x/react-data-grid/accessibility).

Special attention should be taken to keyboard navigation

**Search keywords**:
